### ISSUE TITLE
Problems downloading DRM Protected Content

### ISSUE THREAD

**Issue by misterscraper** [datawhores/OF-Scraper#206]
I'm having a terrible issue with downloading purchased content thats protected by DRM. I have the CDRM setup correctly via keysdb

![SCR-20230914-rzuk](https://github.com/datawhores/OF-Scraper/assets/33800433/553a36d6-5fd3-46fd-9aeb-e159c2c911c1)

Anyone else having such issues? I've tried uninstalling, deleting the .config folder, still the same issue. Still stuck with the temp file names.

Running on Mac OS with all the latest updates and python.

**Comment by misterscraper** [comment]
UPDATE:

I just ran the following command:

ofscraper --log debug

had to run it twice, but it seems to have worked and downloaded these videos correctly, hope this helps anyone running into issues.

**Comment by naaitsab** [comment]
Having the same issue with the audio and video part files. Looked into the logs seems to be an issue with the W1devine key server? [downloadbatch.key_helper_cdrm2:829]  Cannot connect to host 172.106.17[dot]134:8080 ssl:<ssl.SSLContext object at 0x00000240A148BB40> [The semaphore timeout period has expired]

Could be a temporary outage or it's gone.

**Comment by Puk0** [comment]
> cdrm2

¿qué le pusiste en "key-mode-default": ?

**Comment by Puk0** [comment]
> UPDATE:
> 
> I just ran the following command:
> 
> ofscraper --log debug
> 
> had to run it twice, but it seems to have worked and downloaded these videos correctly, hope this helps anyone running into issues.

No se terminaron de descargar porque te marca "error login", solo vuelve a generar otra cookies 
y ponle key mode : 
keydb (con la api) 
o 
manual (con device_private_key y device_client_id_blob)

**Comment by orithecapper** [comment]
Hey, for me everything works fine in version 3.3.3

python -m pip install -U ofscraper==3.3.3


and change to manual 
![image](https://github.com/datawhores/OF-Scraper/assets/131520229/1c20e545-6421-4467-9c4f-8c43fe8dbc7a)

        "private-key": "/home/ori/src/dumper/private_key.pem",
        "client-id": "/home/ori/src/dumper/client_id.bin",
        "key-mode-default": "manual",


IMPORTENT

        "backend": "aio",
        "download-sems": 5,
        "maxfile-sem": 1,


![image](https://github.com/datawhores/OF-Scraper/assets/131520229/084e979f-ea7b-4a12-972e-e9cd63e05184)

also use
ofscraper -u kitalovexoxovip --after 2000 --dupe (this one seems to work fine)
or 
ofscraper -u kitalovexoxovip --after 2000

**Im able to use it**


Also ANDROID 8 L3 CERTIFICATES ARE NO LONGER WORKING 

----

https://of-scraper.gitbook.io/of-scraper/cdm-options

https://forum.videohelp.com/threads/408031-Dumping-Your-own-L3-CDM-with-Android-Studio?s=f22ce643aaaca3496c59a76a2a78c567

(If you want you can add me on telegram @tgsneed , I will help you dump)

if you wanna dump your own certificates 
use 
git clone https://github.com/lollolong/dumper

DONT FORGET --cdm-version
python3 dump_keys.py --cdm-version '16.1.0'

anyways I can give you an L3 certificate, it still works for me after a day 


(Here is my |private_key.pem|client_id.bin)
https://seyarabata.com/650418ab466ca#filename=working_ANDROID_12_L3.zip

-----------



**THE LATEST VERSIONS ARE BROKEN , USE 3.3.3**

---------------

for torrent archives here is what you can use 
##Software used 
# https://github.com/amietn/vcsi/tree/master/vcsi
# (Supports DRM widevine dumping, with manual CDM) https://github.com/datawhores/OF-Scraper 

##thumbs
vcsi *.mp4 --fast -o ../THUMBS/


for f in Posts/Videos/*; do  ffmpeg -i $f -vcodec h264_nvenc -movflags +faststart -crf 23 compressed/${f##*/} -y; done

**Comment by naaitsab** [comment]
https://forum.videohelp.com/threads/408031-Dumping-Your-own-L3-CDM-with-Android-Studio
Could this also help to get the certificates? New to this stuff but I presume you need a signed cert pair to use the Python CDM script as the current key site seems dead?

**Comment by orithecapper** [comment]
> https://forum.videohelp.com/threads/408031-Dumping-Your-own-L3-CDM-with-Android-Studio Could this also help to get the certificates? New to this stuff but I presume you need a signed cert pair to use the Python CDM script as the current key site seems dead?

Use mine 

>(Here is my |private_key.pem|client_id.bin)
>https://seyarabata.com/650418ab466ca#filename=working_ANDROID_12_L3.zip


they are fine until they expire idk lol

**Comment by orithecapper** [comment]
> > > https://forum.videohelp.com/threads/408031-Dumping-Your-own-L3-CDM-with-Android-Studio Could this also help to get the certificates? New to this stuff but I presume you need a signed cert pair to use the Python CDM script as the current key site seems dead?
> > 
> > 
> > Use mine
> > > (Here is my |private_key.pem|client_id.bin)
> > > https://seyarabata.com/650418ab466ca#filename=working_ANDROID_12_L3.zip
> > 
> > 
> > they are fine until they expire idk lol
> 
> Confirmed this works using the wiki: https://of-scraper.gitbook.io/of-scraper/cdm-options Thanks.

Did it help?

**Comment by sivra-d** [comment]
I'm having this problem on both 3.3.3 and the latest version with keysdb. I think it's safe to say this is not currently a valid option for auth, the service seems to just error out consistently[1] unless I have something misconfigured[2], and considering the other auto auth options are down then dumping keys manually seems to be the only way. It would be nice to have something simpler than booting up a virtual android device, though, since:

I'm also unable to get the AVD method liked above (https://forum.videohelp.com/threads/408031-Dumping-Your-own-L3-CDM-with-Android-Studio) for dumping keys to function. All the correct libraries are running, but OF site seems to hang in the emulator despite other sites working fine. I used the same settings in the guide which was a Pixel 3 on Android 9. I haven't tried other configurations since they seem much more involved based on comments in the forum thread.

I appreciate you offering your keys @orithecapper but it's definitely a stop-gap. This stuff isn't my wheelhouse though and so I can't really contribute to a solution beyond being a guinea pig and debugging.

[1][of-debug.log](https://github.com/datawhores/OF-Scraper/files/12678181/of-debug.log)

[2]
```
        "key-mode-default": "keydb",
        "keydb_api": "[my_api_key]",
        "dynamic-mode-default": "deviint",
        "partfileclean": false,
        "backend": "aio",
        "download-sems": 5,
        "maxfile-sem": 1,
```

**Comment by sivra-d** [comment]
Alright, I feel both a bit stupid and also a little justified in the above comment. The AVD method above _does_ work properly, it just doesn't work if you try to hit OF directly. Using the bitmovin demo site to dump keys works fine. 

I still think keysdb is broken, though.

Tested on v3.3.3

**Comment by Puk0** [comment]
> I still think keysdb is broken, though.

si funciona 
https://keysdb.net/api 
![2023-09-21 03-45-53](https://github.com/datawhores/OF-Scraper/assets/138377242/7307d1ee-a41b-4d74-b8fa-fa1c07247bbc)

¿como le pusiste en "keydb_api": "",  api o el KID?

tienes windows usa la versión zip o .exe
![2023-09-21 03-53-56](https://github.com/datawhores/OF-Scraper/assets/138377242/bd108633-df32-43f6-94b6-724934d3e212)

**Comment by orithecapper** [comment]
> keydb

use manual-mode 
[with ](https://of-scraper.gitbook.io/of-scraper/cdm-options)

**Comment by sivra-d** [comment]
Yes, I got manual mode working with my own keys. I still think keydb auto mode is broken (and if it really is broken, and not just me, it should be removed from the documentation).

**Comment by Puk0** [comment]
> > > keydb
> > 
> > 
> > use manual-mode https://seyarabata.com/650418ab466ca#filename=working_ANDROID_12_L3.zip
> 
> any chance you can share another key?

No es una key
Son los archivos cdm

**Comment by dougless12** [comment]
> > > > keydb
> > > 
> > > 
> > > use manual-mode https://seyarabata.com/650418ab466ca#filename=working_ANDROID_12_L3.zip
> > 
> > 
> > any chance you can share another key?
> 
> No es una key Son los archivos cdm

and those cdm files together make a decryption key so yes it is a key
